### DISCUSSION TITLE
Is it possible to group a table by accessor attribute?

### DISCUSSION THREAD

**Discussion by joshembling** [filamentphp/filament#9262] [General]
If i have the following on a table: 

```php
return $table
    ->groups([
        Group::make('featured'),
    ])
```

This may relate to an attribute cast I have on my model: 

```php
public function featured(): Attribute
{
    return Attribute::make(
        get: fn () => $this->started > now() && $this->ended < now()
    );
}
```

However, when doing so I'm getting a `SQLSTATE[42S22]: Column not found: 1054 Unknown column 'featured' in 'order clause'` error when trying to access this attribute. 

Is this currently possible in v3?

**Comment by sinnbeck** [comment]
Maybe try setting the key directly to `$record->featured` 
https://filamentphp.com/docs/3.x/tables/grouping#setting-a-group-key
And just use `Group::make('started')`or similar

**Reply by joshembling** [reply]
Unfortunately, this does not work.

**Comment by sergejostir** [comment]
Create a virtual column.
